zhaquirks/legrand/cable_outlet.py

```diff
@@ -86,87 +127,65 @@ class LegrandCableOutletCluster(CustomCluster):
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Identify.cluster_id,
-                    Groups.cluster_id,
-                    OnOff.cluster_id,
-                    Scenes.cluster_id,
-                    MANUFACTURER_SPECIFIC_CLUSTER_ID,
-                    ElectricalMeasurement.cluster_id,
-                    MANUFACTURER_SPECIFIC_CLUSTER_ID_2,
-                ],
-                OUTPUT_CLUSTERS: [
-                    OnOff.cluster_id,
-                    Basic.cluster_id,
-                    MANUFACTURER_SPECIFIC_CLUSTER_ID,
-                    Scenes.cluster_id,
-                    Ota.cluster_id,
-                ],
-            },
-            #  <SimpleDescriptor endpoint=242 profile=41440 device_type=102
-            # input_clusters=[33]
-            # output_clusters=[33]>
-            242: {
-                PROFILE_ID: zgp.PROFILE_ID,
-                DEVICE_TYPE: zgp.DeviceType.COMBO_BASIC,
-                INPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
-                OUTPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
-            },
-        },
-    }
-
-    replacement = {
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: zha.DeviceType.LEVEL_CONTROL_SWITCH,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Identify.cluster_id,
-                    Groups.cluster_id,
-                    OnOff.cluster_id,
-                    Scenes.cluster_id,
-                    LegrandCluster,
-                    ElectricalMeasurement.cluster_id,
-                    LegrandCableOutletCluster,
-                ],
-                OUTPUT_CLUSTERS: [
-                    OnOff.cluster_id,
-                    Basic.cluster_id,
-                    LegrandCluster,
-                    Scenes.cluster_id,
-                    Ota.cluster_id,
-                ],
-            },
-            242: {
-                PROFILE_ID: zgp.PROFILE_ID,
-                DEVICE_TYPE: zgp.DeviceType.COMBO_BASIC,
-                INPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
-                OUTPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
-            },
-        },
-    }
+    async def write_attributes(self, attributes, manufacturer=None) -> list:
+        """Write attributes to the cluster."""
+
+        attrs = {}
+        for attr, value in attributes.items():
+            attr_def = self.find_attribute(attr)
+            attr_id = attr_def.id
+            if attr_id == HEAT_MODE_ATTR:
+                await self.set_heat_mode(value, manufacturer=manufacturer)
+            else:
+                attrs[attr] = value
+        return await super().write_attributes(attrs, manufacturer)
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == HEAT_MODE_ATTR:
+            self.endpoint.device.thermostat_bus.listener_event(
```

Comment:
I think we can remove this and the device class and just call heat mode change directly from here